frontend/src/utils/account-send.ts

```diff
@@ -0,0 +1,144 @@
+import { isAddress, keccak256, toUtf8Bytes, BigNumber, hexZeroPad } from 'src/utils/ethers';
+import localforage from 'localforage';
+import { toAddress, lookupAddress } from 'src/utils/address';
+import { LOCALFORAGE_ACOUNT_SEND_KEY } from 'src/utils/constants';
+import { Web3Provider } from 'src/utils/ethers';
+
+type AccountData = {
+  address: string;
+  advancedMode: boolean;
+  checkbox: boolean;
+};
+
+type KeyData = {
+  count: number;
+  viewingKey: string;
+};
+
+type EncryptedData = {
+  encryptedAddress: string;
+};
+
+type AccountSendData = {
+  amount: string;
+  tokenAddress: string;
+  dateSent: Date;
+  hash: string;
+  userAddress: string;
+  recipientAddress: string;
+  checkbox?: boolean;
+  advancedMode?: boolean;
+};
+
+type DecryptedSendData = {
+  recipientId: string;
+};
+
+const accountData = ({ address, advancedMode, checkbox }: AccountData) => {
+  if (!isAddress(address)) {
+    throw new Error('Invalid address');
+  }
+  const advancedModeBit = advancedMode ? '1' : '0';
+  const checkboxBit = checkbox ? '1' : '0';
+  const addressHex = BigNumber.from(address).toHexString() + advancedModeBit + checkboxBit;
```

Comment:
As you have it, the `advancedModeBit` and `checkboxBit` are really nibbles (half bytes), not bits, because you're using these strings as a single hex character, and each hex character is a nibble. That's fine since we have extra space, so maybe just make them one byte each for simplicity
```suggestion
  const advancedModeByte = advancedMode ? '01' : '00';
  const checkboxByte = checkbox ? ? '01' : '00';
  const addressHex = BigNumber.from(address).toHexString() + advancedModeByte + checkboxByte;
```